src/core_read_parameter_file.c

```diff
@@ -202,55 +203,77 @@ int read_parameter_file(const int ThisTask, const char *fname, struct params *ru
         return FILE_NOT_FOUND;
     }
 
-    char buffer[MAX_STRING_LEN];
-    while(fgets(&(buffer[0]), MAX_STRING_LEN, fd) != NULL) {
-        char buf1[MAX_STRING_LEN], buf2[MAX_STRING_LEN];
-        if(sscanf(buffer, "%s%s%*[^\n]", buf1, buf2) < 2) {
-            continue;
-        }
-
-        if(buf1[0] == '%' || buf1[0] == '-') {
-            continue;
-        }
-
-        int j=-1;
-        for(int i = 0; i < NParam; i++) {
-            if(strcmp(buf1, ParamTag[i]) == 0) {
-                j = i;
-                ParamTag[i][0] = 0;
-                used_tag[i] = 0;
-                break;
+    if(fd != NULL) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Condition 'fd!=NULL' is always true](https://app.codacy.com/manual/jacobseiler/sage-model/pullRequest?prid=4620074)